### ISSUE TITLE
Xiaomi violates GPL licenses

### ISSUE THREAD

**Issue by sleptworld** [MiCode/Xiaomi_Kernel_OpenSource#4092]
**Kernel Sources Unreleased Models:**
- Redmi Note 11 / NFC (spes/spesn)
- Redmi Note 11 Pro 5G / Redmi Note 11E Pro (veux)
- Xiaomi 12T (plato)
- Xiaomi 12T Pro / Redmi K50 Ultra (diting)
- Redmi Note 12 4G (tapas/topaz)
- Redmi Note 12 5G (sunstone)
- Redmi Note 12 Pro 4G (sweet_k6a)
- Redmi Note 12S (sea)
- ~~Redmi K60 / POCO F5 Pro (mondrian)~~
- Redmi K60E (rembrandt)
- Xiaomi 12 Pro Dimensity Edition (daumier)

**Comment by ddeevvaa** [comment]
Xiaomi violates GPL licenses

**Comment by satishs22** [comment]
https://xiaomifirmwareupdater.com/

can try this

**Comment by cuynu** [comment]
K60 https://github.com/MiCode/Xiaomi_Kernel_OpenSource/tree/mondrian-s-oss

**Comment by cuynu** [comment]
Xiaomi 12 Pro
https://github.com/MiCode/Xiaomi_Kernel_OpenSource/tree/daumier-s-oss

**Comment by sleptworld** [comment]
> Xiaomi 12 Pro https://github.com/MiCode/Xiaomi_Kernel_OpenSource/tree/daumier-s-oss


Ok,  i will remove these items from the list.

**Comment by tptnam** [comment]
and 11t (agate)

**Comment by popy2k14** [comment]
+1 for 11T (agate/amber) @mi-code

**Comment by ech0devv** [comment]
@mi-code 12t

**Comment by Sushrut1101** [comment]
+ Redmi 10A
+ viva
+ evergo-s-oss

**Comment by jstsdev** [comment]
i need Redmi Note 12 4G (tapas/topaz) please

**Comment by lvctr** [comment]
One more for the list:

- Redmi Note 12 Turbo / Poco F5 (marble)

**Comment by KuroShion** [comment]
11T Agate not been release till now

**Comment by str4fe** [comment]
spes/spesn now is released

**Comment by TalasDigital** [comment]
12S Ultra sources are still missing after only 10months. Is there any hope?

**Comment by AnyOne06** [comment]
Diting released.

**Comment by frickypickle** [comment]
Xiaomi 12 still missing (cupid)

**Comment by popy2k14** [comment]
It's a shame that we have to upgrade because of missing source code. Those mobiles are good to go a few years.

Xiaomi, come on.

**Comment by frickypickle** [comment]
> It's a shame that we have to upgrade because of missing source code. Those mobiles are good to go a few years.
> 
> Xiaomi, come on.

Just buy a pixel and don't deal with this shit

**Comment by popy2k14** [comment]
@eclipsek20 That's one reason not too buy an xiaomi again!

**Comment by AnyOne06** [comment]
Xiaomi best

**Comment by frickypickle** [comment]
> @eclipsek20 That's one reason not too buy an xiaomi again!

Yeah I'll go with a pixel or nothing

**Comment by cuynu** [comment]
Not just xiaomi, most manufacturer from china are the same, like Lenovo and ODMs like SDMC, QIN, HISENSE, they are violates GPL critical and NEVER release kernel source. China is the most country that violations almost any GPL rules, they only only only want CLOSED SOURCE

**Comment by newlund** [comment]
The following kernel source is incomplete. Should be added to the list again.

Redmi K60 / POCO F5 Pro (mondrian)

**Xiaomi please don't violate the GPL and support the community. Show that you are not a parasite company.**

**Comment by blacklightpy** [comment]
Redmi Note 12S missing main camera drivers (gotta verify, I just observed that `sea` ROM wasn't building with camera support)

**Comment by popy2k14** [comment]
I bought an Pixel 8 Pro!
Never ever Xiaomi Smartphones!

**Comment by gartriss** [comment]
Redmi Note 10 5G (camellia), incomplete and broken kernel source.

**Comment by madurapa** [comment]
Still not available for Redmi 10A

**Comment by wahyuief** [comment]
Redmi K50 Gaming/Poco F4 GT (ingres) too

**Comment by Become-ILLUSORY** [comment]
redmi k50 pro still missing

**Comment by cornshiba** [comment]
Xiaomi 12S Ultra, as well. Sources are still missing after 2 years and 10 months.

**Comment by SyedHasan110** [comment]
The only solution for this problem is everybody stop buying xiaomi phones.

**Comment by madurapa** [comment]
> The only solution for this problem is everybody stop buying xiaomi phones.

100% agreed!

This is how they make profit: https://www.androidauthority.com/remove-ads-xiaomi-miui-1019139/

**Comment by SyedHasan110** [comment]
Yes, Exactly they earn money using ads but at the cost of destroying user UI/UX Experience of HyperOS. Plus Integrating bloatware shit this makes user annoyed and the user turn towards custom rom for bloatware free and clean UI/UX. Current, I have Installed HyperOS Indonesin version and there are two messaging apps and several unwanted apps that i do not use.

Now, this is the time that we all should make a resolution to never ever use xiaomi phones and waste our precious money on this shit. 

https://github.com/NothingOSS 

The above link is of NothingOSS repo where they upload their devices kernel sources for even mediatek chipset and are regularly updated.

**Comment by madurapa** [comment]
Xiaomi justifies making phone rooting extremely difficult by claiming it's for theft prevention, but the reality is that they primarily want to prevent users from removing bloatware. I'm currently using a Redmi 10A, which will be both my first and last Xiaomi device. Based on this experience, I would never recommend Xiaomi products to anyone.

**Comment by SyedHasan110** [comment]
Same here and I will buy Nothing phone.

**Comment by madurapa** [comment]
Based on my research, Nokia or Motorola are great options if you're on a budget. 

For flagship devices, I’d recommend Nothing or Pixel. 

I’ve used Nokia 6, Pixel 3, and Pixel 5, all fantastic phones that I absolutely loved!

`Though long discontinued, my all-time favorite remains the YotaPhone, a true pioneer with its dual-screen design. RIP to an underrated legend!`

**Comment by Zaptyp** [comment]
Share cupid kernel sources because is near a end of life for getting a major android updates!

**Comment by madurapa** [comment]
If they don't provide the source, team up with [gpl-violations.org](http://gpl-violations.org/faq/violation-faq/). You have the legal right to get the source, because the Linux kernel is under the GPLv2!
